airflow/providers/smtp/notifications/templated_smtp.py

```diff
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Iterable
+
+from airflow.configuration import conf
+from airflow.providers.smtp.notifications.smtp import SmtpNotifier
+from airflow.settings import SMTP_DEFAULT_TEMPLATED_HTML_CONTENT_PATH, SMTP_DEFAULT_TEMPLATED_SUBJECT
+
+# Read once and cache - is this desirable? If users edit local settings, their changes will not take effect
+# On the other hand, if we put this in the constructor, then we need to read from disk on each instantiation
+SMTP_DEFAULT_TEMPLATED_HTML_CONTENT = Path(SMTP_DEFAULT_TEMPLATED_HTML_CONTENT_PATH).read_text()
+
+
+class TemplatedSmtpNotifier(SmtpNotifier):
```

Comment:
I really like the idea of templates!

However I think we should try to keep the current user API.
meaning that users should still use `SmtpNotifier`.
Thus when users do:

`MyOperator(on_success_callable=SmtpNotifier(to='myemail@myemail.com),...)`
It will use the default template of success unless user overrides it

`MyOperator(on_success_callable=SmtpNotifier(to='myemail@myemail.com), template='my_template.html'...)`

Similarly if user does:

`MyOperator(on_failure_callable=SmtpNotifier(to='myemail@myemail.com),...)`
It will use the default template of failure unless user overrides it

Reply:
@hussein-awala , @potiuk - what do you think about @eladkal 's idea? Asking because you both suggested something slightly different in https://github.com/apache/airflow/issues/35381#issuecomment-1791741821 and https://github.com/apache/airflow/issues/35381#issuecomment-1792108885

I'm fine either way.